Add carrier information in case of WiFi connection with SIM card.

Hemant21KG opened the pull request:
Fixed : https://github.com/open-telemetry/opentelemetry-android/issues/1435

Hemant21KG committed:
Add carrier information in case of WiFi connection with SIM card availability

breedx-splk commented:
@Hemant21KG I owe you a huge apology! I have no good/reasonable explanation for why this PR was left unreviewed and and without comment for so long. 

This is a nice contribution and I think we would like to proceed with it, but right now it will need a rebase. If you can do that, that would be awesome. Alternatively, feel free to create a new PR or give us the go ahead to recreate this on your behalf if you don't have time to come back to it.

Once again, I'm so sorry that it took this long. This is not a normal thing here, it's pretty embarrassing, and I'm sincerely sorry. Hopefully we can work it out and get this merged.

Hemant21KG commented:
@breedx-splk Thanks for following up, and no worries at all. I’ve rebased the branch and updated the PR.

Please let me know if there’s anything else that needs to be addressed. Looking forward to getting this merged.

Thanks again for taking the time to review it!

DavidGrath commented on instrumentation/network/src/main/java/io/opentelemetry/android/instrumentation/network/internal/detector/NetworkDetectorImpl.kt:
```diff
@@ -76,7 +76,13 @@ internal class NetworkDetectorImpl(
      * Builds a network for non-cellular networks.
      */
-    private fun buildNetwork(networkState: NetworkState) = CurrentNetwork(networkState)
+    private fun buildNetwork(networkState: NetworkState): CurrentNetwork {
+        val carrier = carrierFinder.get()
+        return CurrentNetwork(
+            state = networkState,
+            carrier = carrier,
+        )
+    }
```
@breedx-splk , I'm still thinking of that edge case where the WiFi could be degraded and both cellular and WiFi are simultaneously active. Would there be a need to actually include that or is this okay as it is?

Hemant21KG replied:
Good point. I think this is okay as-is for now since we're only capturing the current active network

Hemant21KG replied:
@DavidGrath

DavidGrath replied:
True. I think it's okay. I just didn't want it to slip by since I remembered a recent discussion about it. Non blocking

breedx-splk replied:
Yeah, I think this relates to #1772 but isn't exactly the same thing. Until that's addressed, I think this is still an improvement. 👍🏻
